### DISCUSSION TITLE
CMD_TEXT not showing up after CMD_ROTATEAROUND

### DISCUSSION THREAD

**Discussion by jacobbenlewis** [RudolphRiedel/FT800-FT813#112] [Q&A] [answered]
Hello!  This is happening for me in both the EVE Screen Editor and when using this library to actually run the code on my board.  So it's not a problem with the library, but I'm hoping I can find help here.

I am trying to load an image from flash, then rotate it, and that all works.  But when I try to add text after that operation the text doesn't show up.  This is an example of the commands out of EVE Screen Editor

// I can see this text just fine
COLOR_RGB(241, 28, 0)
CMD_TEXT(300, 150, 30, 0, "Hello")

// This part works
CMD_FLASHREAD(0, 8512, 18496)
CMD_SETBITMAP(0, COMPRESSED_RGBA_ASTC_8x8_KHR, 272, 272)
BEGIN(BITMAPS)
CMD_LOADIDENTITY()
VERTEX2F(4233, 3225)
CMD_ROTATEAROUND(136, 136, 16384, 65536)
CMD_SETMATRIX();
VERTEX2F(1744, 5472)
END()

// If after that I try to add:
COLOR_RGB(1, 28, 250)
CMD_TEXT(250, 100, 30, 0, "Goodbye")

The text does not show up.  I can move it above the FLASHREAD and it works.  If I take out either the CMD_ROTATEAROUND or CMD_SETMATRIX it works.

Any help much appreciated.

**Comment by RudolphRiedel** [comment, marked as answer]
The transformation matrix gets applied to the text as well.
A pair of SAVE_CONTEXT() / RESTORE_CONTEXT() calls around the rotation changes that.
EVE_cmd_dl(DL_SAVE_CONTEXT); / EVE_cmd_dl(DL_RESTORE_CONTEXT);

**Reply by jacobbenlewis** [reply]
Ah!  So the text was getting rotated out of view?  I'm still confused by the transform stuff, haven't found a good description of how that works.

But yes, adding the SAVE/RESTORE_CONTEXT took care of my problem, thank you!

**Reply by RudolphRiedel** [reply]
The transformation matrix applies to all images and the individual characters are images, so every single one is rotated.
I have never properly tried to understand what this transformation matrix stuff is actually about, but as far as I know a couple of things in the graphics engine are derived from OpenGL.
